### PR TITLE
[AIRFLOW-2834] fix build script for k8s docker

### PR DESCRIPTION
**Pull request by yeluolei**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-2834
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

the kubernetes docker build airflow without rbac support, but the configmap need rbac. so need to change the build script to build js and css files. 
currently when open airflow web ui deployed in kubernetes, the webpage is blank and will be some file missing.

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [ ] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by yeluolei** [comment]
the travis build failed at check license, seams not a problem.

**Review by ashb — CHANGES_REQUESTED** [review]
The tests are failing as a result of the change though - it's complaining about all the new files that node has created.

You will need to add suitable excludion rules into the `.rat-excludes` file.

**Comment by yeluolei** [comment]
wired, the test complains the files in airflow/www_rbac/node_modules/ 
but there already a rule in .rat-excludes with airflow/www_rbac/node_modules/ 

I have tried with different approach to make the exclude rule work, but failed. can you provide a simple instruction to how to ignore files under airflow/www_rbac/node_modules/  ?

**Comment by yeluolei** [comment]
@ashb

**Comment by samdjstephens** [comment]
@ashb @yeluolei I just went through the pain of trying to use `scripts/ci/kubernetes` to test the kubernetes executor out locally, debugging it and finding this PR. Can we get this merged so no-one else has to go through this again?

**Comment by Fokko** [comment]
@verdan PTAL

File: .rat-excludes [hunk 1]
```diff
@@ -39,5 +39,6 @@ bootstrap-toggle.min.js
 bootstrap-toggle.min.css
 d3.v3.min.js
 ace.js
-airflow/www_rbac/node_modules
+.*node_modules.*
 .*json
+flake8_diff.sh
```
Comment (ashb) [inline .rat-excludes:44]: I don't think these changes should be in this PR as they are unrelated - (also `flake8-diff.sh` doesn't exist anymore)

**Comment by ashb** [comment]
Are these changes needed for running tests, or are you re-using the images built there for running Airflow in kube?

File: .rat-excludes (hunk 1 above)
Comment (ashb) [reply .rat-excludes:44]: Oh I just saw **my own** previous comment. Sorry.

Still, flake8_diff.sh doesn't exist anymore

**Comment by ashb** [comment]
The only reason I hesitate as this will slow down the Travis builds a bit, and they are already slow enough, and this is not needed for running the tests.

Would a comment, or a flag you pass  `build.sh` (i.e. `build.sh --with-assets`) be enough, so that by default on Travis we don't run `compile_assets`, but it could be re-used for your local docker builds? (The primary purpose of the script is for CI, not general purpose)

**Comment by ashb** [comment]
Oh, and additionally/instead:

If we do leave this in we should work out to add .node_modules to the Travis cache.

**Comment by yeluolei** [comment]
@ashb sorry, I'm not very familiar with travis, this change is to make sure our kube change and kube unit test can pass through, and it's four month ago after I made this change...  what do you mean by add some flag to cache? and how to add cache into travis?

**Comment by ashb** [comment]
> this change is to make sure our kube change and kube unit test can pass

What unit tests depend on the front-end assets?

**Comment by kppullin** [comment]
I ran across this same issue and created a duplicate PR: https://github.com/apache/incubator-airflow/pull/4248

While I understand these scripts are for CI, I also found using the Dockerfile and the kube deploy CI script convenient for testing & iterating on changes made to the kubernetes executor, etc.

**Comment by ashb** [comment]
Couldn't you get even quicker dev cycles by volume-mounting the airflow code in? rather than having to rebuild?

**Comment by kppullin** [comment]
That'd likely work well.  I could use the `deploy.sh` script to bootstrap the system in minikube (postgresql, secrets, etc) and then change the airflow deployment to point a volume.

**Comment by yeluolei** [comment]
@ashb is that means we should not use this Dockerfile to build docker image of airflow, instead we should create one by ourself and then use it in k8s cluster? then I think we need to close this issue and then add some readme in the Dockerfile or some where

**Comment by ashb** [comment]
Fixed by #3770

**Pull request closed without merging**